### PR TITLE
[#1151] morph/client: cache notary transaction heights

### PR DESCRIPTION
**Pull request by fyrchik**
Signed-off-by: Evgenii Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: pkg/morph/client/constructor.go [hunk 1]
```diff
@@ -99,6 +102,13 @@ func New(key *keys.PrivateKey, endpoint string, opts ...Option) (*Client, error)
 	}, nil
 }
 
+func initClientCache() cache {
+	lru, _ := lru.New(100) // panics only if size isn't negative
```
Comment (carpawell) [inline pkg/morph/client/constructor.go:106]: "panics", "isn't negative"?

File: pkg/morph/client/notary.go [hunk 2]
```diff
@@ -879,10 +879,22 @@ func (c *Client) CalculateNonceAndVUB(hash util.Uint256) (nonce uint32, vub uint
 	// TODO: #1151 cache values since some operations uses same TX as triggers
```
Comment (carpawell) [inline pkg/morph/client/notary.go:879]: delete this line and link to https://github.com/nspcc-dev/neofs-node/issues/1151? or this PR is not about it?

**Review by carpawell — CHANGES_REQUESTED** [review]

**Review by alexvanin — APPROVED** [review]

File: pkg/morph/client/constructor.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/constructor.go]: still did not get why the "panics" word was used

**Review by carpawell — APPROVED** [review]

File: pkg/morph/client/constructor.go (hunk 1 above)
Comment (fyrchik) [reply pkg/morph/client/constructor.go]: Oh, got that. Fixed.

**Pull request merged**